frontend/packages/dev-console/src/components/import/image-search/ImageSearchSection.tsx

```diff
@@ -1,13 +1,55 @@
 import * as React from 'react';
+import { useFormikContext, FormikValues } from 'formik';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { RadioButtonField } from '../../formik-fields';
 import FormSection from '../section/FormSection';
+import ImageStream from './ImageStream';
 import ImageSearch from './ImageSearch';
 import SearchStatus from './SearchStatus';
 import SearchResults from './SearchResults';
 
-const ImageSearchSection: React.FC = () => {
+export interface ImageSearchSectionProps {
+  projects: K8sResourceKind[];
+  imageStreams: K8sResourceKind[];
+}
+const ImageSearchSection: React.FC<ImageSearchSectionProps> = ({ projects, imageStreams }) => {
+  const { values, setFieldValue, initialValues } = useFormikContext<FormikValues>();
+  const [registry, setRegistry] = React.useState(values.registry);
+
+  React.useEffect(() => {
+    if (values.registry !== registry) {
+      setRegistry(values.registry);
+      setFieldValue('searchTerm', initialValues.searchTerm);
+      setFieldValue('isi', initialValues.isi);
+      setFieldValue('imageStream', initialValues.imageStream);
+    }
+  }, [
+    initialValues.imageStream,
+    initialValues.isi,
+    initialValues.searchTerm,
+    registry,
+    setFieldValue,
+    values,
+  ]);
+
   return (
-    <FormSection title="Image Search" subTitle="Deploy an existing image from an image registry.">
-      <ImageSearch />
+    <FormSection
+      title="Image"
+      subTitle="Deploy an existing image from an image stream or image registry."
+    >
+      <RadioButtonField
+        name="registry"
+        label="Image Section"
+        options={[
+          { label: 'Image name from external registry', value: 'external' },
+          { label: 'Image name from internal registry', value: 'internal' },
+        ]}
+      />
+      {values.registry === 'external' ? (
```

Comment:
I feel we can have an enum instead of writing `'external'`
```
export enum imageRegistry {
  External = 'external',
  Internal = 'internal',
};

or

export const imageRegistry = {
  External: {
    value: 'external',
    label: 'Image name from external registry',
  },
  Internal: {
    value: 'internal',
    label: 'Image name from internal registry',
  },
};